perceval/backends/git.py

```diff
@@ -632,25 +660,46 @@ def log(self, from_date=None, encoding='utf-8'):
 
         :raises RepositoryError: when an error occurs fetching the log
         """
+
         cmd_log = ['git', 'log', '--raw', '--numstat', '--pretty=fuller',
                    '--decorate=full', '--all', '--reverse', '--topo-order',
                    '--parents', '-M', '-C', '-c', '--remotes=origin']
-
         if from_date:
             dt = from_date.strftime("%Y-%m-%d %H:%M:%S %z")
             cmd_log.append('--since=' + dt)
+        env = {'LANG' : 'C', 'PAGER' : ''}
 
-        gitlog = self._exec(cmd_log, cwd=self.dirpath,
-                            env={'LANG' : 'C', 'PAGER' : ''})
-        gitlog = gitlog.split(b'\n')
-
+        self.failed_message = None
```

Comment:
As a general rule, try to add more blank lines to your code. Now it looks like a block. There are some parts that you can split like paragraphs and will make the code easier to read :-D. For instance, I usually add a blank lines between statements and `if` or `for` blocks. For instance, I think this is better:

```python

err_thread.start()

for line in self.proc.stdout:
    yield line.decode('utf-8', errors='surrogateescape')

err_thread.join()

self.proc.communicate()
self.proc.stdout.close()
self.proc.stderr.close()

if self.proc.returncode != 0:
    cause = "git command - %s (return code: %d)" % \
        (self.failed_message, self.proc.returncode)
    raise RepositoryError(cause=cause)
    
logging.debug("Git log fetched from %s repository (%s)",
              self.uri, self.dirpath)
```

than this:

```python
err_thread.start()
# Read stdout, yielding to the consumer of this iterator
for line in self.proc.stdout:
    yield line.decode('utf-8', errors='surrogateescape')
# No more stdout, wait for the thread reading stderr, close descriptors
err_thread.join()
# Wait until subprocess is done
self.proc.communicate()
self.proc.stdout.close()
self.proc.stderr.close()
if self.proc.returncode != 0:
    # Subprocess failed, error in failed_message, from stdout
    cause = "git command - %s (return code: %d)" % \
        (self.failed_message, self.proc.returncode)
    raise RepositoryError(cause=cause)
logging.debug("Git log fetched from %s repository (%s)",
              self.uri, self.dirpath)
```